core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -201,6 +204,8 @@ public UpdateSchema renameColumn(String name, String newName) {
     Types.NestedField field = findField(name);
     Preconditions.checkArgument(field != null, "Cannot rename missing column: %s", name);
     Preconditions.checkArgument(newName != null, "Cannot rename a column to null");
+    Preconditions.checkArgument(
+        !renamedNameToId.containsKey(newName), "Cannot rename columns to same name: %s", newName);
```

Comment:
I prefer not to support such usages, it's too complicated and seldom used. This check aims to prevent such usages early:
```
update.rename("a", "b").rename("c", "b")
```